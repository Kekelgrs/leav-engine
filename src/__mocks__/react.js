const react = require('react');
module.exports = {...react, memo: x => x};

export default undefined;
