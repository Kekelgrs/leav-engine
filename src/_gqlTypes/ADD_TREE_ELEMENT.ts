/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {TreeElementInput} from './globalTypes';

// ====================================================
// GraphQL mutation operation: ADD_TREE_ELEMENT
// ====================================================

export interface ADD_TREE_ELEMENT_treeAddElement {
    id: string | null;
    library: string | null;
}

export interface ADD_TREE_ELEMENT {
    treeAddElement: ADD_TREE_ELEMENT_treeAddElement;
}

export interface ADD_TREE_ELEMENTVariables {
    treeId: string;
    element: TreeElementInput;
    parent?: TreeElementInput | null;
}
