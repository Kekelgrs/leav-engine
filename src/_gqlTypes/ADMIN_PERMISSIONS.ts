/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ADMIN_PERMISSIONS
// ====================================================

export interface ADMIN_PERMISSIONS_adminPermissions {
    admin_access_attributes: boolean;
    admin_access_libraries: boolean;
    admin_access_permissions: boolean;
    admin_access_trees: boolean;
    admin_create_attribute: boolean;
    admin_create_library: boolean;
    admin_create_tree: boolean;
    admin_delete_attribute: boolean;
    admin_delete_library: boolean;
    admin_delete_tree: boolean;
    admin_edit_attribute: boolean;
    admin_edit_library: boolean;
    admin_edit_permission: boolean;
    admin_edit_tree: boolean;
}

export interface ADMIN_PERMISSIONS {
    adminPermissions: ADMIN_PERMISSIONS_adminPermissions | null;
}
