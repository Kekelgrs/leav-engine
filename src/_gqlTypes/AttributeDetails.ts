/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {AttributeType, AttributeFormat, PermissionsRelation, ValueVersionMode} from './globalTypes';

// ====================================================
// GraphQL fragment: AttributeDetails
// ====================================================

export interface AttributeDetails_permissionsConf_permissionTreeAttributes {
    id: string;
    linked_tree: string | null;
    label: any | null;
}

export interface AttributeDetails_permissionsConf {
    permissionTreeAttributes: AttributeDetails_permissionsConf_permissionTreeAttributes[];
    relation: PermissionsRelation;
}

export interface AttributeDetails_versionsConf {
    versionable: boolean;
    mode: ValueVersionMode | null;
    trees: string[] | null;
}

export interface AttributeDetails {
    id: string;
    type: AttributeType;
    format: AttributeFormat | null;
    system: boolean;
    label: any | null;
    linked_library: string | null;
    linked_tree: string | null;
    multipleValues: boolean;
    permissionsConf: AttributeDetails_permissionsConf | null;
    versionsConf: AttributeDetails_versionsConf | null;
}
