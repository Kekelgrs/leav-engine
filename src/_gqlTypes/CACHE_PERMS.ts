/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: CACHE_PERMS
// ====================================================

export interface CACHE_PERMS {
    cachePerms: any | null;
}

export interface CACHE_PERMSVariables {
    perms: any;
}
