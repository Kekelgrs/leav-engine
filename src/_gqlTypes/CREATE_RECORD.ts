/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: CREATE_RECORD
// ====================================================

export interface CREATE_RECORD_createRecord {
    id: string;
}

export interface CREATE_RECORD {
    createRecord: CREATE_RECORD_createRecord;
}

export interface CREATE_RECORDVariables {
    library: string;
}
