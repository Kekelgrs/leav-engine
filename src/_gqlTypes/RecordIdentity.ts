/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: RecordIdentity
// ====================================================

export interface RecordIdentity_whoAmI_library {
    id: string;
    label: any | null;
}

export interface RecordIdentity_whoAmI {
    id: string;
    library: RecordIdentity_whoAmI_library;
    label: string | null;
    color: string | null;
    preview: string | null;
}

export interface RecordIdentity {
    whoAmI: RecordIdentity_whoAmI;
}
