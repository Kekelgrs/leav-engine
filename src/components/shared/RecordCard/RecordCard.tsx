import * as React from 'react';
import {withNamespaces, WithNamespaces} from 'react-i18next';
import {localizedLabel} from 'src/utils/utils';
import {RecordIdentity_whoAmI} from 'src/_gqlTypes/RecordIdentity';
import styled, {CSSObject} from 'styled-components';
import RecordPreview from '../RecordPreview';

interface IRecordCardProps extends WithNamespaces {
    record: RecordIdentity_whoAmI;
    style?: CSSObject;
}

function RecordCard({record, style, i18n}: IRecordCardProps): JSX.Element {
    /* tslint:disable:variable-name */
    const Wrapper = styled.div`
        border-left: 5px solid ${record.color || 'transparent'};
        display: flex;
        flex-direction: row;
        ${style}
    `;
    Wrapper.displayName = 'Wrapper';

    const CardPart = styled.div`
        display: flex;
        flex-direction: column;
        justify-content: center;
    `;

    const PreviewWrapper = styled(CardPart)`
        margin: 0 0.8em;
    `;

    const RecordLabel = styled.div`
        font-weight: bold;
    `;

    const LibLabel = styled.div`
        font-weight: normal;
        color: rgba(0, 0, 0, 0.4);
        fontsize: 0.9em;
    `;
    /* tslint:enable:variable-name */

    return (
        <Wrapper className="ui fluid">
            <PreviewWrapper className="ui">
                <RecordPreview label={record.label || record.id} color={record.color} image={record.preview} />
            </PreviewWrapper>
            <CardPart className="ui">
                <RecordLabel>{record.label || record.id}</RecordLabel>
                <LibLabel>{localizedLabel(record.library.label, i18n) || record.library.id}</LibLabel>
            </CardPart>
        </Wrapper>
    );
}

export default withNamespaces()(RecordCard);
