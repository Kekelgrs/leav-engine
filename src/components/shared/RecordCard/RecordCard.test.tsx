import {shallow} from 'enzyme';
import * as React from 'react';
import {RecordIdentity_whoAmI} from 'src/_gqlTypes/RecordIdentity';
import RecordCard from './RecordCard';

describe('RecordCard', () => {
    test('Snapshot test', async () => {
        const mockRecord: RecordIdentity_whoAmI = {
            id: '12345',
            library: {
                id: 'test_lib',
                label: {fr: 'Test Lib'}
            },
            label: 'Test Record',
            color: null,
            preview: null
        };
        const comp = shallow(<RecordCard record={mockRecord} />);

        expect(comp).toMatchSnapshot();
    });
});
