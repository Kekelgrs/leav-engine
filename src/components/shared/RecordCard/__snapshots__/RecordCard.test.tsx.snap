// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordCard Snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecordCard
    i18n={
        Object {
            "language": "fr",
            "options": Object {
              "fallbackLng": Array [
                "en",
              ],
            },
          }
    }
    record={
        Object {
            "color": null,
            "id": "12345",
            "label": "Test Record",
            "library": Object {
              "id": "test_lib",
              "label": Object {
                "fr": "Test Lib",
              },
            },
            "preview": null,
          }
    }
    t={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown
          className="ui"
>
          <RecordPreview
                    color={null}
                    image={null}
                    label="Test Record"
          />
</Unknown>,
        <Unknown
          className="ui"
>
          <Unknown>
                    Test Record
          </Unknown>
          <Unknown>
                    Test Lib
          </Unknown>
</Unknown>,
      ],
      "className": "ui fluid",
      "recordColor": null,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <RecordPreview
            color={null}
            image={null}
            label="Test Record"
/>,
          "className": "ui",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": null,
            "image": null,
            "label": "Test Record",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
    display: flex;
    flex-direction: column;
    justify-content: center;
",
              "
    margin: 0 0.8em;
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-htpNat",
          ],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown>
              Test Record
</Unknown>,
            <Unknown>
              Test Lib
</Unknown>,
          ],
          "className": "ui",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Test Record",
            },
            "ref": null,
            "rendered": "Test Record",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "
    font-weight: bold;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Test Lib",
            },
            "ref": null,
            "rendered": "Test Lib",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": true,
                "rules": Array [
                  "
    font-weight: normal;
    color: rgba(0, 0, 0, 0.4);
    fontsize: 0.9em;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
    display: flex;
    flex-direction: column;
    justify-content: center;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
    border-left: 5px solid ",
          [Function],
          ";
    display: flex;
    flex-direction: row;
    ",
          [Function],
          "
",
        ],
      },
      "displayName": "Wrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown
            className="ui"
>
            <RecordPreview
                        color={null}
                        image={null}
                        label="Test Record"
            />
</Unknown>,
          <Unknown
            className="ui"
>
            <Unknown>
                        Test Record
            </Unknown>
            <Unknown>
                        Test Lib
            </Unknown>
</Unknown>,
        ],
        "className": "ui fluid",
        "recordColor": null,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <RecordPreview
              color={null}
              image={null}
              label="Test Record"
/>,
            "className": "ui",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": null,
              "image": null,
              "label": "Test Record",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
    display: flex;
    flex-direction: column;
    justify-content: center;
",
                "
    margin: 0 0.8em;
",
              ],
            },
            "displayName": "Styled(styled.div)",
            "foldedComponentIds": Array [
              "sc-htpNat",
            ],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown>
                Test Record
</Unknown>,
              <Unknown>
                Test Lib
</Unknown>,
            ],
            "className": "ui",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Test Record",
              },
              "ref": null,
              "rendered": "Test Record",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": true,
                  "rules": Array [
                    "
    font-weight: bold;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Test Lib",
              },
              "ref": null,
              "rendered": "Test Lib",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-EHOje",
                  "isStatic": true,
                  "rules": Array [
                    "
    font-weight: normal;
    color: rgba(0, 0, 0, 0.4);
    fontsize: 0.9em;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-EHOje",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
    display: flex;
    flex-direction: column;
    justify-content: center;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
    border-left: 5px solid ",
            [Function],
            ";
    display: flex;
    flex-direction: row;
    ",
            [Function],
            "
",
          ],
        },
        "displayName": "Wrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
